extensions/cornerstone/src/init.js

```diff
@@ -54,6 +56,23 @@ export default function init({ servicesManager, configuration }) {
   };
 
   initCornerstoneTools(defaultCsToolsConfig);
+  // TODO: Extensions are still registered at time of registration globally
+  // These should be registered as a part of mode route spin up,
+  // and they need to self-clean on mode route destroy
+  // THIS
+  // is a way for extensions that "depend" on this extension to notify it of
+  // new cornerstone enabled elements so it's commands continue to work.
+  const handleOhifCornerstoneEnabledElementEvent = function(evt) {
+    const { viewportIndex, enabledElement } = evt.detail;
+
+    console.log(evt, viewportIndex, enabledElement);
+    setEnabledElement(viewportIndex, enabledElement);
+  };
+
+  document.addEventListener(
+    'ohif-cornerstone-enabled-element-event',
+    handleOhifCornerstoneEnabledElementEvent
+  );
```

Comment:
The 'ohif-cornerstone-enabled-element-event' is a way for extensions that extend the Cornerstone Viewport in this extension to notify it that a `cornerstone Enabled Element` event occurred, so we can track its reference, so commands continue to function (this is a bit nuts, and I've left some comments regarding clean-up)